app/controllers/support_interface/provider_users_controller.rb

```diff
@@ -84,6 +84,19 @@ def toggle_notifications
       redirect_to support_interface_provider_user_path(provider_user)
     end
 
+    def update_notifications
+      render_404 unless FeatureFlag.active?(:configurable_provider_notifications)
+
+      provider_user = ProviderUser.find(params[:provider_user_id])
+      notification_preferences = provider_user.notification_preferences
+      notification_preferences.assign_attributes(notification_preferences_params)
+
+      if notification_preferences.save!
```

Comment:
Similar to above, I think we can merge these two steps to an `update!` as it will throw a `ActiveRecord::RecordInvalid` similar to `save!`